src/libs/auth.ts

```diff
@@ -399,74 +424,54 @@ export const isAzureUser = (): boolean => {
+    // according to IdTokenClaims, this is a mandatory claim and should always exist
+    const userId: string = tokenClaims.sub;
     return {
       ...state,
-      isSignedIn,
-      anonymousId: !isSignedIn && state.isSignedIn ? undefined : state.anonymousId,
-      registrationStatus: isSignedIn ? state.registrationStatus : 'uninitialized',
-      termsOfService: initializeTermsOfService(isSignedIn, state),
-      profile: isSignedIn ? state.profile : {},
-      nihStatus: isSignedIn ? state.nihStatus : undefined,
-      fenceStatus: isSignedIn ? state.fenceStatus : {},
+      isSignedIn: true,
+      isSigningIn: false,
       // Load whether a user has input a cookie acceptance in a previous session on this system,
       // or whether they input cookie acceptance previously in this session
-      cookiesAccepted: isSignedIn
-        ? state.cookiesAccepted || getLocalPrefForUserId(userId, cookiesAcceptedKey)
-        : undefined,
-      isTimeoutEnabled: isSignedIn ? state.isTimeoutEnabled : undefined,
-      hasGcpBillingScopeThroughB2C: isSignedIn ? state.hasGcpBillingScopeThroughB2C : undefined,
+      cookiesAccepted: state.cookiesAccepted || getLocalPrefForUserId(userId, cookiesAcceptedKey),
       user: {
-        token: user?.access_token,
-        scope: user?.scope,
+        token: user.access_token,
+        scope: user.scope,
         id: userId,
-        ...(profile
-          ? {
-              email: profile.email,
-              name: profile.name,
-              givenName: profile.given_name,
-              familyName: profile.family_name,
-              imageUrl: profile.picture,
-              idp: profile.idp,
-            }
-          : {}),
+        email: tokenClaims.email,
+        name: tokenClaims.name,
+        givenName: tokenClaims.given_name,
+        familyName: tokenClaims.family_name,
+        imageUrl: tokenClaims.picture,
+        idp: tokenClaims.idp,
       },
     };
   });
 };
 
-const initializeTermsOfService = (isSignedIn: boolean, state: AuthState): TermsOfServiceStatus => {
-  return {
-    userHasAcceptedLatestTos: isSignedIn ? state.termsOfService.userHasAcceptedLatestTos : undefined,
-    permitsSystemUsage: isSignedIn ? state.termsOfService.permitsSystemUsage : undefined,
-  };
-};
-
-export const initializeAuth = _.memoize(async () => {
+// this function is only called when the browser refreshes
+export const initializeAuth = _.memoize(async (): Promise<void> => {
   // Instantiate a UserManager directly to populate the logged-in user at app initialization time.
-  // All other auth usage should use the AuthContext from authStore.
+  // All other auth usage should use the AuthContext from oidcStore.
   const userManager: UserManager = new UserManager(getOidcConfig());
-  processUser(await userManager.getUser(), false);
+
+  try {
+    const initialOidcUser: OidcUser | null = await userManager.getUser();
+    if (initialOidcUser !== null) {
+      userLoadedEventHandler(initialOidcUser);
```

Comment:
Nit: since this is function is not only an event handler (it's called directly here), maybe a different name would be more appropriate.

Reply:
We ended up going with `loadOidcUser` so it isn't thought of only as an event handler (although it may make sense to change that if we get rid of `initializeAuth`)